### ISSUE TITLE
Remove fields functionality is not called

### ISSUE THREAD

**Issue by phptek** [silverstripe/silverstripe-restfulserver#85]
Neither `DataFormatter::setRemoveFields()` nor `DataFormatter::getRemoveFields()` is called or referenced from anywhere.

Any fix for this should allow for the default returned field "ID" to be removed (See `DataFormatter::getFieldsForObj():329`.

**Comment by emteknetnz** [comment]
We could possibly look to deprecate these methods rather than outright remove them

Presumably though they're currently not doing any harm or presenting a security risk right now by being there?

**Comment by phptek** [comment]
Correct. But for those of us who use the source code for docs, it's misleading to have them. Maybe mark as @unused or @deprecated?
